index.js

```diff
@@ -25,13 +25,31 @@ var UNICODE_CHARS = {
     '\u2029': '\\u2029'
 };
 
+// We can‘t just instanceof Date since dates are already converted to strings
+// because of native Date.prototype.JSON (which use toISOString)
+var DATE_LENGTH = new Date().toISOString().length
+function isDate(d) {
+  try {
+    return (
+      // testing length first to avoid new Date() for every string
+      d.length === DATE_LENGTH &&
```

Comment:
I think for performance reason, we should avoid regex. That will be very expensive to use regex on all strings without a fast and easy way to detect non date string. Obviously, we need to ensure a date is valid. That's why I mentionned using `new Date(string).getTime()` which is not throwing error (from my test) but that return `NaN` for invalid date.